### DISCUSSION TITLE
npm package latest version not seen

### DISCUSSION THREAD

**Discussion by augi** [renovatebot/renovate#16750] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.126.2

### Please select which platform you are using if self-hosting.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.3.8

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Renovate doesn't create PRs for npm packages. For example, there is `3.8.2` version of chart.js available (we have `3.8.0`) but PR is not created.

When running renovate with debug logging, we can see it fetches information from our NPM source (Artifactory) and that the Artifactoy URL returns the expected JSON (so `3.8.2` is included in the versions).

Please note that we performed the update from 2.x to 3x manually but it shouldn't be any problem.
![image](https://user-images.githubusercontent.com/273052/180752951-a420c82a-0c2b-46fa-b064-85160bc4ebaa.png)

Also, `npm outdated` recommends the new version (so I believe it shouldn't be a problem with our environment).
![image](https://user-images.githubusercontent.com/273052/180754147-30b1955d-2856-4f7b-9bab-4204aefae802.png)

### Relevant debug logs

<details><summary>Logs</summary>

```
 INFO: Dependency extraction complete (repository=project/ui)
       "baseBranch": "master",
       "stats": {
         "managers": {
           "kubernetes": {"fileCount": 1, "depCount": 4},
           "npm": {"fileCount": 2, "depCount": 48}
         },
         "total": {"fileCount": 3, "depCount": 52}
       }

               {
                 "depType": "dependencies",
                 "depName": "chart.js",
                 "currentValue": "^3.8.0",
                 "datasource": "npm",
                 "prettyDepType": "dependency",
                 "lockedVersion": "3.8.0",
                 "depIndex": 4,
                 "updates": [],
                 "warnings": [],
                 "versioning": "npm",
                 "sourceUrl": "https://github.com/chartjs/Chart.js",
                 "homepage": "https://www.chartjs.org",
                 "currentVersion": "3.8.0",
                 "fixedVersion": "3.8.0"
               },

DEBUG: http statistics (repository=project/ui)
       "urls": {
          "https://artifactory.our.company.com/artifactory/api/npm/registry.npmjs.org/chart.js (GET,200)": 1,
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
Not a bug. Look up rangeStrategy in the docs.

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#rangestrategy

**Reply by augi** [reply]
Hello, thank you for your prompt reply! 👍 

According to the log, renovate doesn't take `3.8.2` into account (check `currentVersion` in the log provided) so the `rangeStragery` doesn't play any role. The log provided is DEBUG starting with `packageFiles with updates`.

**Reply by viceice** [reply]
`packageFiles with updates` won't show any in-range update
